ggml-impl.h

```diff
@@ -106,6 +106,9 @@ typedef uint16_t ggml_fp16_internal_t;
 #ifdef _MSC_VER
 #define GGML_COMPUTE_FP16_TO_FP32(x) _mm_cvtss_f32(_mm_cvtph_ps(_mm_cvtsi32_si128(x)))
 #define GGML_COMPUTE_FP32_TO_FP16(x) _mm_extract_epi16(_mm_cvtps_ph(_mm_set_ss(x), 0), 0)
+//If we have F16C, testing shows it's much faster than using the lookup tables.
+#define GGML_FP16_TO_FP32(x) GGML_COMPUTE_FP16_TO_FP32(x)
+#define GGML_FP32_TO_FP16(x) GGML_COMPUTE_FP32_TO_FP16(x)
 #else
```

Comment:
Should this be put right outside of the MSVC check? `_MSC_VER` is only defined by MSVC, so this optimization is only compiled with the Visual Studio compiler. It doesn't look compiler-specific, so maybe this wasn't intended.

Reply:
No, I only have proof inside of MSVC, so I'm not targeting all platforms.